src/key_command/impl_completion.rs

```diff
@@ -0,0 +1,72 @@
+use super::constants::*;
+use super::{Command, Completion};
+
+pub enum CompletionKind<'a> {
+    Bin,
+    Custom(Vec<&'a str>),
+    Dir(Option<Vec<&'a str>>),
+    File,
+}
+
+impl Completion for Command {
+    fn completion_kind(cmd: &str) -> Option<CompletionKind> {
```

Comment:
I initially tried to match on `self`, but some commands need specific conditions to be successfully built via `from_str`, so matching just on the keyword seems to be the only solution. So this could have been a standalone function in `ui/view/tui_textfield.rs`, but I think that having it here as a trait of `Command` makes it more discoverable for future contributors.

Reply:
Yeah, this looks great :+1: